### ISSUE TITLE
older yi camera's dont work with 3.0

### ISSUE THREAD

**Issue by Cadsters** [AlexxIT/WebRTC#391]
I have got two kind of Yi camera's after updating to 3.0 my older one''s with the adress:
```
rtsp://<IP-Adress>:554/ch0_0.h264
```
do not work, while the newer ones with adress:
```
rtsp://<IP-Adress>/ch0_0.h264
```
What could cause this?  They work oké in 2.3.1

btw i have setup go2rtc in docker and got this in my log:
```
06:48:19.658 ERR [api.mjpeg] add consumer error="codecs not match: H264,AAC"

06:48:36.523 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:132 > error="RTSP wrong input" url=rtsp://192.168.2.11:554/ch0_0.h264
```

**Comment by AlexxIT** [comment]
I need access to stream https://github.com/AlexxIT/go2rtc/wiki/Tunnel-RTSP-camera-to-Intenet
If you want, you can share it in private msg to contacts on GitHub profile.

**Comment by Cadsters** [comment]
I'm sorry but i cannot share  the stream, i have got som more info if it helps. Used rtsplog, her are the results.
`rtsplog_amd64.exe rtsp://192.168.2.151:554/ch0_0.h264 > out.txt
2023/01/22 12:37:24 RTSP/1.0 200 OK
CSeq: 1
Date: Sun, Jan 22 2023 11:37:31 GMT
Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, PAUSE, GET_PARAMETER, SET_PARAMETER

2023/01/22 12:37:24 RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://192.168.2.151/ch0_0.h264/
Content-Length: 857
Content-Type: application/sdp
Date: Sun, Jan 22 2023 11:37:31 GMT

v=0
o=- 1674387451600359 1 IN IP4 192.168.1.128
s=H.264 Program Stream, streamed by the LIVE555 Media Server
i=ch0_0.h264
t=0 0
a=DevVer:pusher2
a=GroupName:IPCAM
a=NickName:CIF
a=CfgSection:PROG_CHN0
a=tool:LIVE555 Streaming Media v2011.08.13
a=type:broadcast
a=control:*
a=range:npt=0-
a=x-qt-text-nam:H.264 Program Stream, streamed by the LIVE555 Media Server
a=x-qt-text-inf:ch0_0.h264
m=video 0 RTP/AVP 96
c=IN IP4 0.0.0.0
b=AS:4000
a=rtpmap:96 H264/90000
a=control:trackID=1
a=fmtp:96 packetization-mode=1;profile-level-id=64001F;sprop-parameter-sets=Z2QAH6wrUCgC3IA=,aO48MA==
a=framesize:96 1280-720
a=cliprect:0,0,1280,720
m=audio 0 RTP/AVP 97
a=rtpmap:97 mpeg4-generic/8000/2
a=fmtp:97 streamtype=5;profile-level-id=1;cpresent=0;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3;config=1590
a=control:trackID=2
2023/01/22 12:37:24 available tracks: [0xc000092540 0xc000090640]

rtsplog_amd64.exe rtsp://192.168.2.74/ch0_0.h264 > out.txt
2023/01/22 12:40:51 RTSP/1.0 200 OK
CSeq: 1
Date: Sun, Jan 22 2023 11:40:50 GMT
Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, PAUSE, GET_PARAMETER, SET_PARAMETER

2023/01/22 12:40:51 RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://192.168.2.74/ch0_0.h264/
Content-Length: 494
Content-Type: application/sdp
Date: Sun, Jan 22 2023 11:40:50 GMT

v=0
o=- 1674340789223509 1 IN IP4 192.168.2.74
s=Session streamed by "rRTSPServer"
i=ch0_0.h264
t=0 0
a=tool:LIVE555 Streaming Media v2022.10.01
a=type:broadcast
a=control:*
a=range:npt=now-
a=x-qt-text-nam:Session streamed by "rRTSPServer"
a=x-qt-text-inf:ch0_0.h264
m=video 0 RTP/AVP 96
c=IN IP4 0.0.0.0
b=AS:500
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=4D0020;sprop-parameter-sets=Z00AIJZUA8ARLyzcBAQFAAADAfQAAE4ghA==,aO48gA==
a=control:track1
2023/01/22 12:40:51 available tracks: [0xc000092540]`

**Comment by AlexxIT** [comment]
I don't see nothing special in responses. You can increase log level for go2rtc to trace and check something useful there.

**Comment by chex01** [comment]
I have the same problem.

info from go2rtc web:

> {
  "producers": [
    {
      "type": "RTSP source",
      "url": "rtsp://10.10.10.23/ch0_1.h264/",
      "remote_addr": "10.10.10.23:554",
      "user_agent": "go2rtc/1.0.0",
      "medias": [
        "video, sendonly, 96 H264/90000",
        "audio, sendonly, 97 MPEG4-GENERIC/8000/2"
      ],
      "tracks": [
        "96 H264/90000, sinks=4",
        "97 MPEG4-GENERIC/8000/2, sinks=4"
      ],
      "recv": 1394477
    }
  ],
  "consumers": [
    {
      "type": "RTSP client",
      "url": "rtsp://10.10.10.10:8554/Gostinaya",
      "remote_addr": "172.30.33.3:40288",
      "user_agent": "Lavf59.27.100",
      "medias": [
        "video, recvonly, 96 H264/90000",
        "audio, recvonly, 97 MPEG4-GENERIC/8000/2"
      ],
      "tracks": [
        "96 H264/90000, sinks=4",
        "97 MPEG4-GENERIC/8000/2, sinks=4"
      ],
      "send": 1394477
    },
    {
      "type": "RTSP client",
      "url": "rtsp://10.10.10.10:8554/Gostinaya",
      "remote_addr": "10.10.10.10:56880",
      "medias": [
        "video, recvonly, 96 H264/90000",
        "audio, recvonly, 97 MPEG4-GENERIC/8000/2"
      ],
      "tracks": [
        "96 H264/90000, sinks=4",
        "97 MPEG4-GENERIC/8000/2, sinks=4"
      ],
      "send": 1035412
    },
    {
      "type": "MP4 client",
      "remote_addr": "172.30.32.2:39260",
      "user_agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"
    },
    {
      "type": "MP4 client",
      "remote_addr": "172.30.32.2:57162",
      "user_agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"
    }
  ]
}

**Comment by AlexxIT** [comment]
@chex01 RTSP server proxy RTSP stream without any modification. That's why stream works in VLC.

WebRTC and MSE requires stream processing. I think this hacked cameras uses buggy RTSP software with non standard packets. Can't be fixed without access to stream.

**Comment by AlexxIT** [comment]
Fixed in v3.0.1

**Comment by hardwareadictos** [comment]
Same here with tapo C200, v3.0,2:

![image](https://user-images.githubusercontent.com/20403906/218452570-9f66bdd9-1898-4281-833f-0000a3283df5.png)
